### PR TITLE
#5: The solution to the median of two sorted arrays problem

### PR DESCRIPTION
**Pull request by ArinaSiselina**

### CODE REVIEW

File: ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using LeetCodeTraining.MedianOfTwoArrays;
+
+namespace LeetCodeTraining.Tests
+{
+    public class MedianOfTwoArraysSolutionTest
+    {
+        [Theory]
+        [InlineData(new int[]{1, 3}, new int[]{2}, 2.00000)]
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs]: Please align the code. Ctrl+K+F.

File: ScnSoft-Education/LeetCodeTraining/TwoSum1/MedianOfTwoarrays5/MedianOfTwoArraysSolution.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            while (j < nums2.Length)
+            {
+                merged[k++] = nums2[j++];
+            }
+
+            if (n % 2 == 0)
+            {
+                return (merged[n / 2 - 1] + merged[n / 2]) / 2.0;
+            }
+            else
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/TwoSum1/MedianOfTwoarrays5/MedianOfTwoArraysSolution.cs]: Else block is redundant.

File: ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using LeetCodeTraining.MedianOfTwoArrays;
+
+namespace LeetCodeTraining.Tests
+{
+    public class MedianOfTwoArraysSolutionTest
+    {
+        [Theory]
+        [InlineData(new int[]{1, 3}, new int[]{2}, 2.00000)]
+        [InlineData(new int[] { 1, 2 }, new int[] { 3, 4 }, 2.50000)]
+        [InlineData(new int[] { 1, 3 }, new int[] { 5, 8 }, 4.00000)]
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs]: Please add [1,3] [2] from the TestCase tab.

**Review by shomzascnsoft — COMMENTED** [review]
Please move folder with implementation of problem into the root.
image

File: ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace LeetCodeTraining.Tests
+{
+    public class MedianOfTwoArraysSolutionTest
+    {
+        [Theory]
+        [InlineData(new int[] { 1, 3 }, new int[] { 2 }, 2.00000)]
+        [InlineData(new int[] { 1, 2 }, new int[] { 3, 4 }, 2.50000)]
+        [InlineData(new int[] { 1, 3 }, new int[] { 5, 8 }, 4.00000)]
+        public void MedianOfTwoArrays(int[] nums1, int[] nums2, double expectedResult)
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs]: Please add Sorted in the method name.

File: ScnSoft-Education/LeetCodeTraining/MedianOfTwoarrays5/MedianOfTwoArraysSolution.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            }
+
+            if (n % 2 == 0)
+            {
+                return (merged[n / 2 - 1] + merged[n / 2]) / 2.0;
+            }
+            return merged[n / 2];
+        }
+    }
+}
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/MedianOfTwoarrays5/MedianOfTwoArraysSolution.cs:41]: Please update the screenshot to see all submissions, the code and details with diagram results ()Memory, Runtime).
Zoom in the page if it is required to see this on one page.

File: ScnSoft-Education/LeetCodeTraining/Program.cs [hunk 6]
```diff
@@ -1,2 +1,2 @@
 ﻿// See https://aka.ms/new-console-template for more information
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/Program.cs:1]: Please revert the changes in this file if it is possible.
Changes in it not intendent to be in this PR.

File: ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+﻿using LeetCodeTraining.MedianOfTwoArrays;
+
+namespace LeetCodeTraining.Tests
+{
+    public class MedianOfTwoArraysSolutionTest
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs]: Please rename the class to be Test at the and to consistency with other test classes.

Comment (shomzascnsoft) [reply ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs]: The comment above still not fixed. By the class I meant class file name. Sorry for not precision.

File: ScnSoft-Education/LeetCodeTraining/Program.cs (hunk 6 above)
Comment (shomzascnsoft) [reply ScnSoft-Education/LeetCodeTraining/Program.cs:1]: Ok, leave this changes but try to not add not related to the PR changes.
Please do them in main branch if it is required.

File: ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs (hunk 7 above)
Comment (ArinaSiselina) [reply ScnSoft-Education/LeetCodeTraining.Tests/MedianOfTwoArraysTests.cs]: Got it. Now its renamed.

File: ScnSoft-Education/LeetCodeTraining/MedianOfTwoarrays5/MedianOfTwoArraysSolution.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+            while (j < nums2.Length)
+            {
+                merged[k++] = nums2[j++];
+            }
+
+            if (n % 2 == 0)
+            {
+                return (merged[n / 2 - 1] + merged[n / 2]) / 2.0;
+            }
+            67return merged[n / 2];
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/MedianOfTwoarrays5/MedianOfTwoArraysSolution.cs]: Please always check that solution is built and unit test can be performed before requesting the code review and doing push.

Please remove 67.

**Review by shomzascnsoft — CHANGES_REQUESTED** [review]

**Review by shomzascnsoft — APPROVED** [review]
